plugins/funder.c

```diff
@@ -486,18 +486,66 @@ static bool previously_reserved(struct bitcoin_outpoint **prev_outs,
 	return false;
 }
 
+struct funder_utxo {
+	struct bitcoin_outpoint out;
+	struct amount_sat val;
+};
+
+static struct out_req *
+build_utxopsbt_request(struct command *cmd,
+		       struct open_info *info,
+		       struct amount_sat requested_funds,
+		       struct amount_sat committed_funds,
+		       struct funder_utxo **avail_utxos)
+{
+	struct out_req *req;
+
+	req = jsonrpc_request_start(cmd->plugin, cmd,
+				    "utxopsbt",
+				    &psbt_funded,
+				    &psbt_fund_failed,
+				    info);
+	/* Add every prev_out */
+	json_array_start(req->js, "utxos");
+	for (size_t i = 0; i < tal_count(info->prev_outs); i++)
+		json_add_outpoint(req->js, NULL, info->prev_outs[i]);
+
+	/* Next add available utxos until we surpass the
+	 * requested funds goal */
+	/* FIXME: Update `utxopsbt` to automatically add more inputs? */
```

Comment:
Could be an extra RPC method called `fundpsbt` to mirror `bitcoind`'s API.

Reply:
We already have a `fundpsbt` that behaves slightly differently.